src/soap-module-options.type.ts

```diff
@@ -1,38 +1,50 @@
 import { IOptions } from 'soap';
 import { ModuleMetadata, Scope, Type } from '@nestjs/common/interfaces';
-import { BASIC_AUTH, WSSECURITY_AUTH } from './soap-constants';
+import { BASIC_AUTH, WSSECURITY_AUTH, SSL_PFX_AUTH } from './soap-constants';
 
 export { Client, IOptions } from 'soap';
 
 interface Auth {
-  type: typeof BASIC_AUTH | typeof WSSECURITY_AUTH;
+  type: typeof BASIC_AUTH | typeof WSSECURITY_AUTH | typeof SSL_PFX_AUTH;
   username: string;
```

Comment:
I believe the `username` and `password` should be turned into optional now, since it seems the SSL PFX doesn't require them.